Add milliseconds to SQLite datetimes

ItachiSan opened the pull request:
The SQLite DATETIME() method does not provide milliseconds nor has a switch for toggling them.
This is overall inconvenient.
By calling correctly STRFTIME(), we can have the milliseconds information appropriately added.

In addition, the CI configuration file was updated to support newer Python and SQLAlchemy versions.

ItachiSan committed:
Add milliseconds to SQLite datetimes

The SQLite DATETIME() method does not provide milliseconds nor has a
switch for toggling them.
This is overall inconvenient.
By calling correctly STRFTIME(), we can have the milliseconds
information appropriately added.

In addition, the CI configuration file was updated to support newer
Python and SQLAlchemy versions.

dahlia commented on .travis.yml:
```diff
@@ -6,12 +6,17 @@ python:
 - 3.4
 - 3.5
 - 3.6
+- 3.7
+- 3.8
+- 3.9
```
Could you add those versions to _tox.ini_ as well?

ItachiSan replied:
Absolutely!
Should I rebase my commits or are multiple commits fine?

dahlia replied:
Multiple commits are fine!

dahlia reviewed (commented):
Could you write a test case to ensure if the `utcnow()` expression within SQLite dialect emits the precise `datetime.datetime` value?

ItachiSan commented:
I can definitely do that! How much of an overhead should I have? 0.5 seconds?

ItachiSan committed:
Multiple CI / verification improvements

- Tox environments for newer Python versions have been added
- Big improvements in Travis CI
  - Not testing agains EOL SQLAlchemy 0.9
  - Fetch correct dependencies versions for older Python envs
- Temporarily adjust the README for my personal branch, plus point to
  newer Travis CI (from .org it has unified in .com)

ItachiSan committed:
Further tests & dependencies adjustments

- Avoid testing older Python versions with SQLAlchemy v1.4 as no package
  is available for them
- Resolve the issue with the newer `pytest-cov` not working with older
  `pytest` by installing an older version where necessary

ItachiSan committed:
Fix AND operator in check

The double square brackets in Bash uses '&&' while the single square
brackets uses '-a'.
I essentially used the wrong convention here :P

dahlia commented on README.rst:
```diff
@@ -3,10 +3,10 @@ SQLAlchemy-Utc
-   :target: https://travis-ci.org/spoqa/sqlalchemy-utc
-.. image:: https://codecov.io/github/spoqa/sqlalchemy-utc/coverage.svg?branch=master
-   :target: https://codecov.io/github/spoqa/sqlalchemy-utc?branch=master
+.. image:: https://travis-ci.com/ItachiSan/sqlalchemy-utc.svg?branch=master
+   :target: https://travis-ci.com/ItachiSan/sqlalchemy-utc
+.. image:: https://codecov.io/github/ItachiSan/sqlalchemy-utc/coverage.svg?branch=master
+   :target: https://codecov.io/github/ItachiSan/sqlalchemy-utc?branch=master
```
Could you remove this change?

ItachiSan replied:
I can revert my username for sure :)
I used that for verifying the Travis-CI and Codecov integration!
travis-ci.org however became .com, so I'll keep that :)

ItachiSan committed:
Mention supported versions

Travis-CI can still keep up even on unsupported Python versions.
Just mention that those are unsupported and add references so that
in the future the deprecated versions can be removed easily.

Also, restore CI links to the official project.

ItachiSan commented:
Hi, is there any issue with the current pull request?
I was expecting this to be merged sooner or later.

dahlia reviewed (approved):
It looks good to me.  @pbzweihander could you please take a look?

pbzweihander reviewed (approved):
Looks good to me, too!

pbzweihander commented:
@Jhuni0123 Could you review and merge this PR?

Jhuni0123 reviewed (approved):
👍  Good. Thanks for your contribution!
